### PR TITLE
Issue/17

### PR DESCRIPTION
**Pull request by rikky0611**
Related issue: #17 

## Change Summary
- I moved `disp_event` and `events_to_mat` under `utils/` because it seemed a bit wired to include them in `read_buffer.cpp`
- I deleted `loop_readData` in `read_buffer.cpp`.  Actually this is a little important change that we shall ask users to write loop function by themselves as written in `examples/buffer_csv/` and `examples/buffer_davis`. But I'm sure this is much more flexible for now.

### CODE REVIEW

File: src/evl/utils/event_utils.cpp [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+}
+
+cv::Mat events_to_mat(std::vector<EventTuple> events) {
+    int W = 240;
+    int H = 180;
+    cv::Mat src = cv::Mat::zeros(H, W, CV_8UC1);
+    for (auto iter = events.begin(); iter != events.end(); ++iter) {
+        int x = std::get<1>(*iter);
+        int y = std::get<2>(*iter);
+        src.at<unsigned char>(y, x) = 255;  // assume binary image
```
Comment (shiba24) [inline src/evl/utils/event_utils.cpp]: I think the output matrix image is upside-down. Can you check it?

File: src/evl/utils/event_utils.cpp [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+    std::get<3>(x) << std::endl;
+}
+
+cv::Mat events_to_mat(std::vector<EventTuple> events) {
+    int W = 240;
+    int H = 180;
+    cv::Mat src = cv::Mat::zeros(H, W, CV_8UC1);
+    for (auto iter = events.begin(); iter != events.end(); ++iter) {
+        int x = std::get<1>(*iter);
+        int y = std::get<2>(*iter);
```
Comment (shiba24) [inline src/evl/utils/event_utils.cpp]: How can it be possible to distinguish `int pol = std::get<3>(*iter);` ?  The column has two values--0/1. Maybe red/blue or, use gray scale?

File: src/evl/utils/event_utils.hpp [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+
+#include <opencv2/core/core.hpp>
+
+#include <evl/core/types.hpp>
+
+void disp_event(EventTuple x); 
+
+cv::Mat events_to_mat(std::vector<EventTuple> events);
+
+#endif    // SRC_EVL_UTILS_EVENT_UTILS_HPP_
```
Comment (shiba24) [inline src/evl/utils/event_utils.hpp]: src/evl/utils/event_utils.hpp:9:  Line ends in whitespace.  Consider deleting these extra spaces.  [whitespace/end_of_line] [4]
src/evl/utils/event_utils.hpp:11:  Add #include <vector> for vector<>  [build/include_what_you_use] [4]

File: src/evl/utils/event_utils.cpp [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+        src.at<unsigned char>(y, x) = 255;  // assume binary image
+    }
+    return src;
+}
```
Comment (shiba24) [inline src/evl/utils/event_utils.cpp]:
```src/evl/utils/event_utils.cpp:14:  Add #include <vector> for vector<>  [build/include_what_you_use] [4]
src/evl/utils/event_utils.cpp:24:  Could not find a newline character at the end of the file.  [whitespace/ending_newline] [5]
```

Comment (shiba24) [inline src/evl/utils/event_utils.cpp]:
```
src/evl/utils/event_utils.cpp:14:  Add #include <vector> for vector<>  [build/include_what_you_use] [4]

src/evl/utils/event_utils.cpp:24:  Could not find a newline character at the end of the file.  [whitespace/ending_newline] [5]
```

File: src/evl/utils/event_utils.hpp (hunk 3 above)
Comment (shiba24) [inline src/evl/utils/event_utils.hpp]:
```
src/evl/utils/event_utils.hpp:9:  Line ends in whitespace.  Consider deleting these extra spaces.  [whitespace/end_of_line] [4]

src/evl/utils/event_utils.hpp:11:  Add #include <vector> for vector<>  [build/include_what_you_use] [4]
```

File: src/evl/utils/event_utils.cpp (hunk 1 above)
Comment (shiba24) [inline src/evl/utils/event_utils.cpp]: I think the image matrix now is upside-down. Please check it.

File: src/evl/utils/event_utils.cpp (hunk 2 above)
Comment (shiba24) [inline src/evl/utils/event_utils.cpp]: how can you distinguish polarity? It has two possible values: 0 and 1.

File: examples/buffer_csv/main.cpp [hunk 5]
```diff
@@ -1,19 +1,47 @@
...
     int lifetime = 1000;     // micro sec
     int buffersize = 50000;
     EventBuffer buffer(buffersize);
-    char fname[] = "../../data/7.csv";
-
+    char fname[] = "../../data/sample.csv";
     std::thread t1(bufferData, &buffer, fname);
-    std::thread t2(loop_readData, &buffer, lifetime);
-
+    //loop_readData(&buffer, lifetime);
```
Comment (shiba24) [inline examples/buffer_csv/main.cpp]: Should have a space between // and comment

File: src/evl/utils/event_utils.cpp (hunk 1 above)
Comment (shiba24) [reply src/evl/utils/event_utils.cpp]: I mean, upside-down when I checked buffer_csv/main.cpp. So maybe because of cv::imshow, or cv::mat itself?

Comment (rikky0611) [reply src/evl/utils/event_utils.cpp]: Hmm, I don't get it. I use tsue sample data and can see in the right way. Which data are you using?

File: src/evl/utils/event_utils.cpp (hunk 4 above)
Comment (rikky0611) [reply src/evl/utils/event_utils.cpp]: fixed

File: src/evl/utils/event_utils.hpp (hunk 3 above)
Comment (rikky0611) [reply src/evl/utils/event_utils.hpp]: fixed

File: examples/buffer_csv/main.cpp (hunk 5 above)
Comment (rikky0611) [reply examples/buffer_csv/main.cpp]: fixed

File: src/evl/utils/event_utils.cpp (hunk 2 above)
Comment (rikky0611) [reply src/evl/utils/event_utils.cpp]: For now, we only consider grayscale image. However, as you pointed out, we need to prepare both cases by adding an argument to indicate which mode to use.  What do you think?

File: src/evl/utils/event_utils.cpp (hunk 1 above)
Comment (shiba24) [reply src/evl/utils/event_utils.cpp]: OK now we realized that something is different between batched-sample file and large sample file (before batch).
I think it's better to check and discuss it in another issue for sample dataset #20 !

File: src/evl/utils/event_utils.cpp (hunk 2 above)
Comment (shiba24) [reply src/evl/utils/event_utils.cpp]: I agree with you, using an argument is a good way. You may create another issue, or implement in this PR, please choose whichever you want!
If creating another issue, you can merge this PR. All other things looks good to me.

Comment (rikky0611) [reply src/evl/utils/event_utils.cpp]: Thanks. I'll open another issue and will solve in the next sprint.

File: src/evl/utils/event_utils.cpp (hunk 1 above)
Comment (muskie82) [reply src/evl/utils/event_utils.cpp]: You should avoid using cv::Mat .at method for value access.
Pointer access is faster.
https://qiita.com/hmichu/items/0a399d9e3bbf3a2a4454

**Pull request merged**